docs/types.rst

```diff
@@ -0,0 +1,506 @@
+
+This example shows adding additional parameters, and explicitly defining the azure_config::
+
+  loadbalancerbackendpool:
+    type: cloudify.azure.nodes.network.LoadBalancer.BackendAddressPool
+    properties:
+      name: mylb
+      location: { get_input: location }
+      retry_after: { get_input: retry_after }
+      azure_config: *azure_config
+    relationships:
+      - type: cloudify.azure.relationships.contained_in_load_balancer
+        target: loadbalancer
+
+
+.. cfy:node:: cloudify.azure.nodes.network.LoadBalancer.Probe
+
+This example shows adding additional parameters, and explicitly defining the azure_config::
+
+  loadbalancerprobe:
+    type: cloudify.azure.nodes.network.LoadBalancer.Probe
+    properties:
+      name: lbprobe
+      location: { get_input: location }
+      retry_after: { get_input: retry_after }
+      azure_config: *azure_config
+      resource_config:
+        protocol: Http
+        port: { get_input: webserver_port }
+        requestPath: index.html
+    relationships:
+    - type: cloudify.azure.relationships.contained_in_load_balancer
+      target: loadbalancer
+    - type: cloudify.relationships.depends_on
+      target: loadbalancerbackendpool
+
+
+.. cfy:node:: cloudify.azure.nodes.network.LoadBalancer.IncomingNATRule
+
+
+.. cfy:node:: cloudify.azure.nodes.network.LoadBalancer.Rule
+
+This example shows adding additional parameters, and explicitly defining the azure_config::
+
+  loadbalancerrule:
+    type: cloudify.azure.nodes.network.LoadBalancer.Rule
+    properties:
+      name: mylbrule
+      location: { get_input: location }
+      retry_after: { get_input: retry_after }
+      azure_config: *azure_config
+      resource_config:
+        protocol: Tcp
+        backendPort: { get_input: webserver_port }
+        frontendPort: { get_input: loadbalancer_port }
+    relationships:
+    - type: cloudify.azure.relationships.contained_in_load_balancer
+      target: loadbalancer
+    - type: cloudify.azure.relationships.connected_to_ip_configuration
+      target: loadbalanceripcfg
+    - type: cloudify.azure.relationships.connected_to_lb_be_pool
+      target: loadbalancerbackendpool
+    - type: cloudify.azure.relationships.connected_to_lb_probe
+      target: loadbalancerprobe
+
+
+.. cfy:node:: cloudify.azure.nodes.storage.DataDisk
+.. cfy:node:: cloudify.azure.nodes.storage.FileShare
+.. cfy:node:: cloudify.azure.nodes.network.NetworkInterfaceCard
+
+
+Node Types Common Behaviour
+===========================
+
+Common Properties
+-----------------
+
+All cloud resource nodes have common properties:
+
+  * ``name``
+  * ``location``
+  * ``tags``
+  * ``retry_after`` Because Azure's API is asynchronous, this value indicates the interval between retries.
+
+Every time you manage a resource with Cloudify,
+we create one or more clients with Azure API.
+You specify the configuration for these clients using the ``azure_config`` property.
+It should be a dictionary,
+with the following values::
```

Comment:
The second part of this change I find slightly jarring ... "using the `azure_config` property as a dictionary ..."
We're not using the property as a dictionary, we're creating the dictionary and providing it to the property.
Would you be happy with
```rst
Specify the configuration for these clients
using the ``azure_config`` property.
It should be a dictionary
with the following values::
```
(extraneous comma removed), or is there another better way to write this?